### PR TITLE
Add `permissions` util and remove permissions-init logic from components

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR follows on from #1685 and removes `permissions` initialization from components and moves it into the `createAnnotation` action in the store, where it finds a better home.

The new `permissions` util is basically a modernized `permissions` service, minus dependencies. I've also added an `isPrivate` function because it seemed useful.

This PR also updates the "modern" (i.e. preact) components to favor the util instead of the service.

The idea here is that the `permissions` service will die as soon as we've migrated remaining uses of it out of the legacy `Annotation` controller—rewriting that code now would be fairly futile as its life expectancy is on the order of days or weeks.

Phew, with this, there is no longer any annotation-object muddling happening in the `annotation` component.

Part of #1650

### CODE REVIEW

File: src/sidebar/components/annotation-omega.js [hunk 1]
```diff
@@ -14,20 +13,9 @@ import AnnotationQuote from './annotation-quote';
 function AnnotationOmega({
   annotation,
   onReplyCountClick,
-  permissions,
   replyCount,
   showDocumentInfo,
 }) {
-  /**
```
Comment (lyzadanger) [inline src/sidebar/components/annotation-omega.js]: Hooray bye bye.

File: src/sidebar/store/modules/test/annotations-test.js [hunk 2]
```diff
@@ -13,7 +15,7 @@ const { actions, selectors } = annotations;
 
 function createTestStore() {
   return createStore(
-    [annotations, selection, drafts, groups, session, viewer],
+    [annotations, selection, defaults, drafts, groups, session, viewer],
```
Comment (lyzadanger) [inline src/sidebar/store/modules/test/annotations-test.js]: I know, I know, I know. We need to solve this `store` dependency/mocking strategy thing ASAP.

Comment (robertknight) [reply src/sidebar/store/modules/test/annotations-test.js]: This code smell is telling me that we should think about a more systematic pattern / rules for handling actions need to operate across multiple domains within the app or which are operations that essentially compose a set of low-level operations. At a minimum, I think we want to separate these operations into a separate module.

I've been having a think on this and will post some ideas later.

There is no need to change anything in this PR though.

File: src/sidebar/store/modules/test/annotations-test.js [hunk 3]
```diff
@@ -448,6 +463,18 @@ describe('sidebar/store/modules/annotations', function() {
       );
     });
 
+    it('should set default permissions on a new annotation', () => {
+      fakeDefaultPermissions.returns('somePermissions');
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/store/modules/test/annotations-test.js]: That seems like a good convention.

File: src/sidebar/util/test/permissions-test.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import * as permissions from '../permissions';
+
+const userid = 'acct:flash@gord.on';
+
+describe('sidebar/util/permissions', () => {
+  describe('#privatePermissions', () => {
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/util/test/permissions-test.js]: You're right—`#` is typically used in JSDoc namepaths for member methods. It is no longer a `Permissions` object, but individual functions here. I'll fix.

**Pull request merged**